cache/metadata/metadata.go

```diff
@@ -94,12 +94,18 @@ func (s *Store) Search(ctx context.Context, index string) ([]*StorageItem, error
 		if main == nil {
 			return nil
 		}
-		index = indexKey(index, "")
+		if !prefix {
+			index = indexKey(index, "")
+		}
 		c := b.Cursor()
 		k, _ := c.Seek([]byte(index))
 		for {
 			if k != nil && strings.HasPrefix(string(k), index) {
-				itemID := strings.TrimPrefix(string(k), index)
+				idx := strings.LastIndex(string(k), "::")
+				if idx == -1 {
+					continue
+				}
+				itemID := string(k[idx+2:])
```

Comment:
should we `if idx+2 < len(k) {` to handle out of bounds?

Reply:
actually that's fine looking at https://github.com/moby/buildkit/pull/5306/files#diff-ae44b7efd0878d4ad4013864764f0046c5c75f5f7dfe08718ad3bd32cabf9caeR104

Reply:
That should not be possible. Then the `idx` would be `-1`. In the worst case `itemID` can be `""`